contracts/src/KeyManager.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+contract KeyManager is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+    struct CommitteeMember {
+        bytes pubKey;
+        bytes secureChannelKey;
+        bytes dkgEncKey;
+        string networkAddress;
+    }
+
+    struct Committee {
+        uint64 effectiveTimestamp;
+        uint64 id;
+        CommitteeMember[] committeeMembers;
+    }
+
+    event ScheduledCommittee(uint64 indexed id, uint64 effectiveTimestamp, uint64 membersCount, bytes32 membersHash, address indexed scheduledBy);
+    event SetThresholdEncryptionKey(bytes thresholdEncryptionKey, address indexed setBy);
+    event ChangedManager(address indexed oldManager, address indexed newManager, address indexed changedBy);
```

Comment:
first, names could be better, 
second, since we already store all those info in the contract storage, we don't need such informative event. 

```suggestion
    event NewCommittee(uint64 indexed id);
    event ThresholdEncKeyUpdated();
    event ManagerChanged(address indexed oldManager, address indexed newManager);
```